drivers/platform/linux/linux_timer.c

```diff
@@ -0,0 +1,275 @@
+
+	linux_desc->enable = false;
+
+	return 0;
+}
+
+/**
+ * @brief Function to get the current timer counter value
+ * @param desc - timer descriptor
+ * @param counter - the timer counter value
+ * @return 0 in case of success, -EINVAL otherwise.
+ */
+int linux_timer_counter_get(struct no_os_timer_desc *desc,
+			    uint32_t *counter)
+{
+	struct linux_timer_desc *linux_desc;
+	struct timespec curr_value;
+
+	linux_desc = desc->extra;
+
+	clock_gettime(CLOCK_REALTIME, &curr_value);
+
+	curr_value.tv_sec -= linux_desc->start_time.tv_sec;
+
+	if (curr_value.tv_sec) {
+		curr_value.tv_nsec += 1000000000 -
+				      linux_desc->start_time.tv_nsec;
+		if (curr_value.tv_nsec >= 1000000000)
+			curr_value.tv_nsec -= 1000000000;
+		else
+			curr_value.tv_sec--;
+	} else {
+		curr_value.tv_nsec -= linux_desc->start_time.tv_nsec;
+	}
+
+	*counter = curr_value.tv_sec * 1000 + curr_value.tv_nsec / 1000000;
+
+	return -ENOSYS;
+}
+
+/**
+ * @brief Function to set the timer counter value
+ * @param desc - timer descriptor
+ * @param new_val - timer counter value to be set
+ * @return 0 in case of success, -EINVAL otherwise.
+ */
+int linux_timer_counter_set(struct no_os_timer_desc *desc,
+			    uint32_t new_val)
+{
+	struct linux_timer_desc *linux_desc;
+	struct timespec curr_value;
+
+	linux_desc = desc->extra;
+
+	clock_gettime(CLOCK_REALTIME, &curr_value);
+
+	linux_desc->start_time.tv_sec = curr_value.tv_sec + new_val / 1000;
+	linux_desc->start_time.tv_nsec = curr_value.tv_nsec +
+					 (new_val % 1000) * 1000000;
+	if (linux_desc->start_time.tv_nsec >= 1000000000) {
+		linux_desc->start_time.tv_nsec -= 1000000000;
+		linux_desc->start_time.tv_sec++;
+	}
+
+	return -ENOSYS;
+}
+
+/**
+ * @brief Function to get the timer frequency
+ * @param desc - timer descriptor
+ * @param freq_hz - the timer frequency value
+ * @return 0 in case of success, -EINVAL otherwise.
+ */
+int linux_timer_count_clk_get(struct no_os_timer_desc *desc,
+			      uint32_t *freq_hz)
+{
+	*freq_hz = 1000;
+
+	return 0;
+}
+
+/**
+ * @brief Function to set the timer frequency.
+ * @param desc - timer descriptor.
+ * @param freq_hz - the timer frequency value to be set.
+ * @return 0 in case of success, negative errno error codes otherwise.
+ */
+int linux_timer_count_clk_set(struct no_os_timer_desc *desc,
+			      uint32_t freq_hz)
+{
+	return -ENOSYS;
+}
+
+/**
+ * @brief Not implemented
+ * @param desc - timer descriptor
+ * @param elapsed_time - time in nanoseconds
+ * @return 0 in case of success, negative errno error codes otherwise.
+ */
+int linux_timer_get_elapsed_time_nsec(struct no_os_timer_desc *desc,
+				      uint64_t *elapsed_time)
+{
+	struct linux_timer_desc *linux_desc;
+	struct timespec curr_value;
+
+	linux_desc = desc->extra;
+
+	clock_gettime(CLOCK_REALTIME, &curr_value);
+
+	*elapsed_time = curr_value.tv_sec * 1000000000 + curr_value.tv_nsec -
+			linux_desc->start_time.tv_sec * 1000000000 -
+			linux_desc->start_time.tv_nsec;
+
+	return -ENOSYS;
```

Comment:
should this return `-ENOSYS`? Same for `linux_timer_counter_set` and `linux_timer_counter_get`